Blueprints/src/BlueprintForm.php

```diff
@@ -387,6 +388,45 @@ protected function deepMerge(array $a, array $b)
         return $a;
     }
 
+
+    /**
+     * @param array $items
+     * @param array $path
+     * @return string
+     */
+    protected function deepImport(array &$items, $path = [])
+    {
+        $run_again = false;
+        $field = end($path) === 'fields';
+
+        foreach ($items as $key => &$item) {
+            if ($field && isset($item['type'])) {
+                $item['name'] = $key;
+            }
+            // Handle special instructions in the form.
+            if (strpos($key, '@') !== false) {
+                // Remove @ from the start and the end. Key syntax `import@2` is supported to allow multiple operations of the same type.
+                $list = explode('-', (string)preg_replace('/^(@*)?([^@]+)(@\d*)?$/', '\2', $key), 2);
+                $action = array_shift($list);
+
+                if ($action == 'import') {
+                    unset($items[$key]);
+                    $this->doImport($item, $path);
```

Comment:
We could do `deepImport()` call inside `doImport()`.

BTW, the lookup scope may change if the file defines its own default lookup location. I think it's somewhat handled inside that method...

Reply:
I simplified `deepImport()`. Don't you think it's good enough to make it in?

If not, do you mind changing the code in a way you prefer, in order to avoid going back and forth about coding perspective?

Thank you